runtime/doc/treesitter.txt

```diff
@@ -678,24 +708,12 @@ inspect_language({lang})          *vim.treesitter.language.inspect_language()*
     Return: ~
         (table)
 
-                                  *vim.treesitter.language.require_language()*
-require_language({lang}, {path}, {silent}, {symbol_name})
-    Asserts that a parser for the language {lang} is installed.
-
-    Parsers are searched in the `parser` runtime directory, or the provided
-    {path}
+register({lang}, {filetype})              *vim.treesitter.language.register()*
+    Register a lang to be used for a filetype (or list of filetypes).
```

Comment:
Is this something that would be done by filetype plugins? 

Any way to make this passive (via some file in 'runtimepath')? Is this a "many to many" mapping?

Reply:
> Is this something that would be done by filetype plugins?

Yes, and also nvim-treesitter / any parser management plugin. But also by the user, e.g: `vim.treesitter.language.register('bash', 'myniche_sh')`.

>  Is this a "many to many" mapping?

No, I want this to be a multiple filetypes to single lang mapping (mentioned in the OP). I really don't think there is a justifiable (or existing) use case for any filetype needing to use multiple parsers at the top level. Every exotic application I've seen would be handled by either: a complicated parser and/or language injections.

> Any way to make this passive (via some file in 'runtimepath')? Instead of "active" state managed at runtime.

The problem with the files in runtimepath method is that it makes it very difficult to redefine/override. I believe this is why `after/` was created. Also, I don't know what a "passive" would look like, though open to suggestions (or discussion). I can only think of a method where maybe `ftplugin` files have some directive to say what parser they should use (maybe setting a future `'lang'` option), and this would be set per buffer as opposed to what I've done here which keeps a separate side table.

Reply:
I think the many to one makes sense here. Many parsers for one filetype are still possible via injections. Our current model has a single root language but injections can even be added for the whole document